### PR TITLE
feat: migrate to Safe version 1.4.1

### PR DESCRIPTION
**Pull request by schmanu**
## How this PR fixes it
Set `LATEST_SAFE_VERSION` to `1.4.1`

This change does not offer a migration transaction for 1.3.0 Safes yet. That will follow in a separate PR.

## How to test it
We need a full regression for this feature as it affects most aspects of the interface.
I so far tested:
- Safe creation for 1.4.1
- Counterfactual Safe creation for 1.4.1
- Counterfactual Safe activation for a previously created 1.3.0 Safe
- Recovery with 1.4.1 Safe
- Recovery with an old 1.3.0 Safe
- Multisends
- Twaps
- Uprading a Safe from 1.1.1 to 1.4.1

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,31 +19,31 @@ Create a `.env` file with environment variables. You can use the `.env.example`
 
 Here's the list of all the environment variables:
 
-| Env variable                                           | Description
```
Comment (schmanu) [inline README.md:22]: This diff was some autoformatting by my VS Code.. If we do not want this I have to revert and Save without formatting.

**Review by katspaugh — APPROVED** [review]
🚀

**Comment by francovenica** [comment]
This ticket is in ready for QA in notion so I tested it

LGTM

Upgrade a safe 1.1.1 to 1.4.1 in ETH mainnet( I don't have one in sepolia)
Recovery in safes 1.3.0
Recovery in safes 1.4.1
CF safes 1.4.1
Deployment or activation of a CF safe just fine (once done the "L2" shows up next to the version
The upgrade of a CF safe 1.3.0 (I had to create a CF safe in dev, copy/paste the local storage to have it in this PR)
Several tx of different kind:
Swap, limit and twaps
Send funds
signing messages
spending limits
owner editting

**Comment by katspaugh** [comment]
Please don’t merge it yet, I’d like to make another release before this. Is it possible to hide it behind a feature toggle?

**Comment by liliya-soroka** [comment]
Safe creation on ZKSync Era doesn't work

File: src/config/chains.ts [hunk 2]
```diff
@@ -1,4 +1,22 @@
 import { networks } from '@safe-global/protocol-kit/dist/src/utils/eip-3770/config'
+import { type SafeVersion } from '@safe-global/safe-core-sdk-types'
+import { getSafeSingletonDeployment } from '@safe-global/protocol-kit/node_modules/@safe-global/safe-deployments'
+import semverSatisfies from 'semver/functions/satisfies'
+import { LATEST_SAFE_VERSION } from './constants'
+
+export const getLatestSafeVersion = (chainId?: string): SafeVersion => {
+  // Without version filter it will always return the LATEST_SAFE_VERSION constant to avoid automatically updating to the newest version if the deployments change
+  const latestVersion = (getSafeSingletonDeployment({ network: chainId, released: true })?.version ??
+    '1.3.0') as SafeVersion
```
Comment (katspaugh) [inline src/config/chains.ts:10]: Let's extract `1.3.0` into a `FALLBACK_VERSION` constant.

Also, this file stopped being a "config" when we switched to generated chain lists. So I would suggest to not make it even less of a config and move this function to utils or services.

**Review by katspaugh — CHANGES_REQUESTED** [review]
Please add a feature toggle so that we can merge this PR any time and toggle a version switch remotely.

**Review by katspaugh — APPROVED** [review]
LGTM but e2e tests are failing

**Comment by francovenica** [comment]
I tried to create a safe in those networks and they don't show the error now. I don't have funds for all those networks, but I was able to create it in zkSycn era and linea with no issues. Safes 1.3.0 were created for those

I tested every network again and none of them showed the error message. so LGTM

**Review by katspaugh — APPROVED** [review]
👍

**Comment by francovenica** [comment]
I've been told by Andrei that the contracts for 1.4.1 safes have been deployed for all networks already. Should then be possible for these few networks where safes are still being created as 1.3.0 to now be created as 1.4.1?

**Comment by schmanu** [comment]
@francovenica you got to enable the feature flag.
I will enable it on all chains except Polygon on staging such that you can also test that Polygon still uses 1.3.0.

**Pull request merged**